source/common/http/header_map_impl.h

```diff
@@ -468,6 +482,11 @@ class RequestHeaderMapImpl final : public TypedHeaderMapImpl<RequestHeaderMap>,
   static std::unique_ptr<RequestHeaderMapImpl> create() {
     return std::unique_ptr<RequestHeaderMapImpl>(new (inlineHeadersSize()) RequestHeaderMapImpl());
   }
+  static std::unique_ptr<RequestHeaderMapImpl> create(const uint32_t max_headers_kb,
+                                                      const uint32_t max_headers_count) {
+    return std::unique_ptr<RequestHeaderMapImpl>(
+        new (inlineHeadersSize()) RequestHeaderMapImpl(max_headers_kb, max_headers_count));
```

Comment:
here and elsewhere: I'm not very familiar with this - is there any potential interaction between inlineHeadersSize() and max_headers_kb?

Reply:
This is a good question.  

I think this is inlineHeaderSize is for custom header, which i am sure it will be part of header map [headers](https://github.com/envoyproxy/envoy/blob/057c80b8215dccc4a7d76e14cb3bb59c91208d6b/source/common/http/header_map_impl.h#L307). 

add @yanavlasov  for comments

Reply:
This inlineHeadersSize() seem to be some extra parameters for this memory malloc.  I am not sure how much it does with the header map memory size. But I think it is not related to the new HCM configuration limits.

Reply:
The inline headers are an O(1) statically allocated header-map, that are stored using InlineStorage.
These are headers that are typically used (e.g., `:authority`, `:path`, and a few common ones), and allows fast access to them. They are always allocated with each header-map, and stored using the `inline_headers_` array.

The config limits are on the number of used headers. So both (used) inline and non-inline headers are counted against these limits.

Reply:
Yes, this is what I meant and want to confirm. So, they are related.
But we don't need to change anything here as size check already covered the inline header.